test/unit/uk/gov/hmrc/testuser/controllers/TestUserControllerSpec.scala

```diff
@@ -16,51 +16,84 @@
 
 package unit.uk.gov.hmrc.testuser.controllers
 
-import org.apache.http.HttpStatus.{SC_INTERNAL_SERVER_ERROR, SC_CREATED}
+import org.apache.http.HttpStatus.{SC_CREATED, SC_INTERNAL_SERVER_ERROR, SC_OK, SC_UNAUTHORIZED}
 import org.mockito.BDDMockito.given
+import org.mockito.Matchers
 import org.scalatest.mock.MockitoSugar
-import play.api.libs.json.Json
-import play.api.test.FakeRequest
+import play.api.libs.json.{JsValue, Json}
+import play.api.test._
 import uk.gov.hmrc.domain._
-import uk.gov.hmrc.play.test.{WithFakeApplication, UnitSpec}
+import uk.gov.hmrc.play.test.{UnitSpec, WithFakeApplication}
 import uk.gov.hmrc.testuser.controllers.TestUserController
 import uk.gov.hmrc.testuser.models.JsonFormatters._
 import uk.gov.hmrc.testuser.models._
-import uk.gov.hmrc.testuser.services.TestUserService
+import uk.gov.hmrc.testuser.repository.TestUserRepository
+import uk.gov.hmrc.testuser.services.{Generator, PasswordService, TestUserService}
 
 import scala.concurrent.Future.failed
 
 class TestUserControllerSpec extends UnitSpec with MockitoSugar with WithFakeApplication {
-  val testIndividual = TestIndividual("user", "password", SaUtr("1555369052"), Nino("CC333333C"))
-  val testOrganisation = TestOrganisation("user", "password", SaUtr("1555369052"), EmpRef("555","EIA000"),
-    CtUtr("1555369053"), Vrn("999902541"))
+
+  val user = "user"
+  val wrongUser = "wrong-usr"
+  val password = "password"
+  val hashedPassword = "_(pwd)"
+  val wrongPassword = "wrong-pwd"
+
+  val saUtr = SaUtr("1555369052")
+  val nino = Nino("CC333333C")
+  val ctUtr = CtUtr("1555369053")
+  val vrn = Vrn("999902541")
+  val empRef = EmpRef("555","EIA000")
+
+  val testIndividual = TestIndividual(user, password, saUtr, nino)
+  val testHashedIndividual = TestIndividual(user, hashedPassword, saUtr, nino)
+
+  val testOrganisation = TestOrganisation(user, password, saUtr, empRef, ctUtr, vrn)
+  val testHashedOrganisation = TestOrganisation(user, hashedPassword, saUtr, empRef, ctUtr, vrn)
 
   trait Setup {
     implicit lazy val materializer = fakeApplication.materializer
 
-    val request = FakeRequest()
+    val createRequest = FakeRequest()
+
+    def authenticationRequest(usr: String, pwd: String) = {
+      val jsonPayload: JsValue = Json.parse(s"""{ "username": "$usr", "password" :"$pwd" }""")
+      FakeRequest().withBody[JsValue](jsonPayload)
+    }
+
     val underTest = new TestUserController {
-      override val testUserService: TestUserService = mock[TestUserService]
+      override val testUserService = new TestUserService {
```

Comment:
we should mock the service in this case as it is a unit tests; and have a unit test for the service